crest/Assets/Crest/Crest/Scripts/Spline/Spline.cs

```diff
@@ -0,0 +1,124 @@
+        public bool _closed = false;
+
+        public SplinePoint[] SplinePoints => GetComponentsInChildren<SplinePoint>();
+    }
+
+#if UNITY_EDITOR
+    public partial class Spline : IValidated
+    {
+        public void OnDrawGizmos()
+        {
+            Gizmos.color = Color.black * 0.5f;
+            var points = SplinePoints;
+            for (int i = 0; i < points.Length - 1; i++)
+            {
+                Gizmos.DrawLine(points[i].transform.position, points[i + 1].transform.position);
+            }
+
+            Gizmos.color = Color.white;
+
+            if (_closed && points.Length > 2)
+            {
+                Gizmos.DrawLine(points[points.Length - 1].transform.position, points[0].transform.position);
+            }
+        }
+
+        public bool Validate(OceanRenderer ocean, ValidatedHelper.ShowMessage showMessage)
+        {
+            var isValid = true;
+
+            var points = SplinePoints;
+
+            if (points.Length < 2)
+            {
+                showMessage
+                (
+                    "Spline must have at least 2 spline points. Click the <i>Add point</i> button in the Inspector, or add a child GameObject and attach <i>SplinePoint</i> component to it.",
+                    ValidatedHelper.MessageType.Error, this
+                );
+
+                isValid = false;
+            }
+            else if (_closed && points.Length < 3)
+            {
+                showMessage
+                (
+                    "Closed splines must have at least 3 spline points. See the <i>Closed</i> parameter and tooltip. To add a point click the <i>Add point</i> button in the Inspector, or add a child GameObject and attach <i>SplinePoint</i> component to it.",
+                    ValidatedHelper.MessageType.Error, this
+                );
+
+                isValid = false;
+            }
+
+            for (int i = 0; i < transform.childCount; i++)
+            {
+                if (transform.GetChild(i).GetComponent<SplinePoint>() == null)
+                {
+                    showMessage
+                    (
+                        $"All child GameObjects under <i>Spline</i> must have <i>SplinePoint</i> component added. Object <i>{transform.GetChild(i).gameObject.name}</i> does not and should have one added, or be moved out of the hierarchy.",
+                        ValidatedHelper.MessageType.Error, this
+                    );
+
+                }
+            }
+
+            return isValid;
+        }
+    }
+
+    [CustomEditor(typeof(Spline))]
+    public class SplineEditor : ValidatedEditor
+    {
+        public override void OnInspectorGUI()
+        {
+            base.OnInspectorGUI();
+
+            var targetSpline = target as Spline;
+
+            if (GUILayout.Button("Add point (extend)"))
+            {
+                SplinePointEditor.AddSplinePointAfter(targetSpline.transform);
```

Comment:
This needs to be recorded with Undo.RegisterCreatedObjectUndo. See other review comments for example.

Reply:
done